File: bin/shell/src/controller/controller.py
from src.controller.builtinController import BuiltinController
from src.controller.databaseController import DatabaseController
from src.controller.collectionController import CollectionController


class Controller:

    @staticmethod
    def handler(block: str) -> None:
        splitBlock = block.split()

        # Catch the illegal block length. The length of the block must be 1 or 2

        if len(splitBlock) == 0 or len(splitBlock) > 2:
            BuiltinController.help()

            return

        # Select the controller to run

        command = splitBlock.__getitem__(0)

        if len(splitBlock) == 1:
            if command == "help":
                BuiltinController.help()
            elif command == "use":
                BuiltinController.use()
            elif command == "name":
                BuiltinController.name()
            elif command == "clear":
                BuiltinController.clear()
            elif command == "exit":
                BuiltinController.exit()
            else:
                print(f"No builtin command named {command} found!")
        else:
            subcommand = splitBlock.__getitem__(1)

            if command == "database":
                if subcommand == "getDatabaseNames":
                    DatabaseController.getDatabaseNames()
                elif subcommand == "getDatabaseSize":
                    DatabaseController.getDatabaseSize()
                elif subcommand == "createDatabase":
                    DatabaseController.createDatabase()
                elif subcommand == "getCollectionNames":
                    DatabaseController.getCollectionNames()
                elif subcommand == "getCollectionSize":
                    DatabaseController.getCollectionSize()
                elif subcommand == "setDatabaseName":
                    DatabaseController.setDatabaseName()
                elif subcommand == "drop":
                    DatabaseController.drop()
                else:
                    print(f"No database command named {subcommand} found!")
            elif command == "collection":
                if subcommand == "createCollection":
                    CollectionController.createCollection()
                elif subcommand == "setCollectionName":
                    CollectionController.setCollectionName()
                elif subcommand == "getAllObjects":
                    CollectionController.getAllObjects()
                elif subcommand == "findFromObjectId":
                    CollectionController.findFromObjectId()
                elif subcommand == "find":
                    CollectionController.find()
                elif subcommand == "insert":
                    CollectionController.insert()
                elif subcommand == "updateFromObjectId":
                    CollectionController.updateFromObjectId()
                elif subcommand == "update":
                    CollectionController.update()
                elif subcommand == "deleteFromObjectId":
                    CollectionController.deleteFromObjectId()
                elif subcommand == "delete":
                    CollectionController.delete()
                elif subcommand == "drop":
                    CollectionController.drop()
                else:
                    print(
                        f"No collection command named {subcommand} found!")
            else:
                print(f"No command named {command} found!")

File: tools/shell/src/main/java/com/padr/tuxdb/shell/controller/Controller.java
package com.padr.tuxdb.shell.controller;

import com.google.gson.JsonSyntaxException;

public class Controller {

    public static void handler(String block) throws JsonSyntaxException, Exception {
        String[] splitBlock = block.split("\\s++");

        if ((splitBlock.length == 1 && splitBlock[0].isEmpty())) {
            BuiltinController.help();

            return;
        }

        String command = splitBlock[0];

        if (splitBlock.length == 1) {
            switch (command) {
                case "help":
                    BuiltinController.help();

                    break;
                case "useDatabase":
                    BuiltinController.useDatabase();

                    break;
                case "useCollection":
                    BuiltinController.useCollection();

                    break;
                case "clear":
                    BuiltinController.clear();

                    break;
                case "exit":
                    BuiltinController.exit();

                    break;
                default:
                    System.out.println(String.format("No builtin command named %s found!", command));

                    break;
            }
        } else {
            String subcommand = splitBlock[1];

            if (command.equals("database")) {
                switch (subcommand) {
                    case "getDatabaseNames":
                        DatabaseController.getDatabaseNames();

                        break;
                    case "getName":
                        DatabaseController.getName();

                        break;
                    case "setName":
                        DatabaseController.setName();

                        break;
                    case "getCollectionNames":
                        DatabaseController.getCollectionNames();

                        break;
                    case "drop":
                        DatabaseController.drop();

                        break;
                    default:
                        System.out.println(String.format("No database command named %s found!", subcommand));

                        break;
                }
            } else if (command.equals("collection")) {
                switch (subcommand) {
                    case "getName":
                        CollectionController.getName();
                        
                        break;
                    case "setName":
                        CollectionController.setName();

                        break;
                    case "getAllObjects":
                        CollectionController.getAllObjects();

                        break;
                    case "findFromObjectId":
                        CollectionController.findFromObjectId();

                        break;
                    case "findOne":
                        CollectionController.findOne();

                        break;
                    case "find":
                        CollectionController.find();

                        break;
                    case "insert":
                        CollectionController.insert();

                        break;
                    case "updateFromObjectId":
                        CollectionController.updateFromObjectId();

                        break;
                    case "update":
                        CollectionController.update();

                        break;
                    case "deleteFromObjectId":
                        CollectionController.deleteFromObjectId();

                        break;
                    case "delete":
                        CollectionController.delete();

                        break;
                    case "drop":
                        CollectionController.drop();

                        break;
                    default:
                        System.out.println(String.format("No collection command named %s found!", subcommand));

                        break;
                }
            }
            else 
                System.out.println(String.format("No command named %s found!", command));
        }
    }
}
